<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Documentation</title>
    <link href="https://unpkg.com/tailwindcss@^1.0/dist/tailwind.min.css" rel="stylesheet" />
    <link rel="stylesheet" href="https://unpkg.com/aos@next/dist/aos.css" />
    <style>
        html {
            scroll-behavior: smooth;
        }
    </style>
</head>

<body>
    <!-- Navigation begins  -->

    <header class="text-gray-600 body-font">
        <div class="container mx-auto flex flex-wrap p-5 flex-col md:flex-row items-center">
            <a class="flex title-font font-medium items-center text-gray-900 mb-4 md:mb-0" href="https://kdsinghcoder.github.io/Align-it">
                <svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="140" height="90">
                  <path fill="#0c7ecf" d="M21.786,80.958l17.784-48.048l7.8-0.39l17.316,48.438l6.474,1.482l0,4.68l-22.62,0l0-4.68l7.254-1.482l-2.964-8.658l-20.28,0l-3.042,8.658l7.332,1.482l0,4.68l-21.45,0l0-4.68z M50.724,66.06l-7.878-23.478l-8.19,23.478l16.068,0z M87.0408,87.12l0-4.68l7.8-1.56l0-42.12l-10.14,0l-1.95,7.8l-5.07,0l0-14.04l42.51,0l0,14.04l-5.07,0l-1.95-7.8l-10.14,0l0,42.12l7.8,1.56l0,4.68l-23.79,0z" viewBox="0 0 24 24"/>
                  <path fill="#" d=""/></svg>
    
                <span class="ml-3 text-xl">- > Align-it</span>
              </a>
            <nav class="md:ml-auto md:mr-auto flex flex-wrap items-center text-base justify-center">
                <a class="mr-5 hover:text-gray-900"></a>
                <a class="mr-5 hover:text-gray-900"></a>
                <a class="mr-5 hover:text-gray-900"></a>
                <a class="mr-5 hover:text-gray-900"></a>
            </nav>
            <a href="../index.html">
                <button
                    class="inline-flex items-center bg-gray-100 border-0 py-1 px-3 focus:outline-none hover:bg-gray-200 rounded text-base mt-4 md:mt-0">Home
                    Page
                    <svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round"
                        stroke-width="2" class="w-4 h-4 ml-1" viewBox="0 0 24 24">
                        <path d="M5 12h14M12 5l7 7-7 7"></path>
                    </svg>
                </button>
            </a>
        </div>
    </header>
    <!-- Navigation ends -->


    <!-- Content begins -->

    
    <div class="container w-full px-5 py-24 mx-auto lg:px-32">
        <div class="flex flex-col w-full mx-auto mb-2 prose text-left prose-md">
          <div class="mb-5 border-b border-gray-200">
            <div class="flex flex-wrap items-baseline -mt-2">
              <h5>12-03-2019</h5>
              <p class="mt-1 ml-2">The Align Library. </p>
            </div>
          </div>
          <div class="aspect-w-16 aspect-h-9 mx-auto my-5">
            <iframe width="640" height="360" src="https://www.youtube.com/embed/9bCkAsaP_z4" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
          </div>
                <h1 class="mb-2 text-4xl font-extrabold leading-snug tracking-tight text-gray-800 md:text-5xl text-gray-900 hover:text-purple-700"> What is Sequence Alignment, and why use it? </h1>
          <p  class="mb-4 text-base font-normal text-gray-600">Sequence alignment is the process of comparing and detecting similarities between biological sequences. What “similarities” are being detected will depend on the goals of the particular alignment process.  </p>
          <p  class="mb-4 text-base font-normal text-gray-600"> If two sequences in an alignment share a common ancestor, mismatches can be interpreted as point mutations and gaps as indels (that is, insertion or deletion mutations) introduced in one or both lineages in the time since they diverged from one another. In sequence alignments of proteins, the degree of similarity between amino acids occupying a particular position in the sequence can be interpreted as a rough measure of how conserved a particular region or sequence motif is among lineages.</p>
          <p  class="mb-4 text-base font-normal text-gray-600"> Alignments are commonly represented both graphically and in text format. In almost all sequence alignment representations, sequences are written in rows arranged so that aligned residues appear in successive columns. In text formats, aligned columns containing identical or similar characters are indicated with a system of conservation symbols. </p>
          <h1 class="mb-2 text-lg font-extrabold leading-snug tracking-tight text-gray-800 md:text-lg text-gray-900 hover:text-purple-700">Types Of Alignment - </h1>
          <p  class="mb-4 text-base font-normal text-gray-600">When two symbolic representations of DNA or protein sequences are arranged next to one another so that their most similar elements are juxtaposed they are said to be aligned. Many bioinformatics tasks depend upon successful alignments. Alignments are conventionally shown as a traces.

In a symbolic sequence each base or residue monomer in each sequence is represented by a letter. The convention is to print the single-letter codes for the constituent monomers in order in a fixed font (from the N-most to C-most end of the protein sequence in question or from 5' to 3' of a nucleic acid molecule). This is based on the assumption that the combined monomers evenly spaced along the single dimension of the molecule's primary structure. From now on we will refer to an alignment of two protein sequences.</p>
          <h3 class="mb-2 text-md font-extrabold leading-snug tracking-tight text-gray-800 md:text-md text-gray-900 hover:text-purple-700"> Pair-wise Alignment and Multiple Sequence Alignment - </h3>
          <p  class="mb-4 text-base font-normal text-gray-600">Pairwise sequence alignment methods are used to find the best-matching piecewise (local or global) alignments of two query sequences. Pairwise alignments can only be used between two sequences at a time, but they are efficient to calculate and are often used for methods that do not require extreme precision (such as searching a database for sequences with high similarity to a query).  Multiple sequence alignment is an extension of pairwise alignment to incorporate more than two sequences at a time. Multiple alignment methods try to align all of the sequences in a given query set. Multiple alignments are often used in identifying conserved sequence regions across a group of sequences hypothesized to be evolutionarily related.</p>
        </div>
      </div>
    

    <!-- Content Ends -->
    
       <!-- Footer starts -->
      
       <footer class="bg-white" aria-labelledby="footer-heading">
        <h2 id="footer-heading" class="sr-only">Footer</h2>
        <div class="px-4 py-12 mx-auto bg-gray-50 max-w-7xl sm:px-6 lg:px-16">
          <div class="flex flex-wrap items-baseline lg:justify-center">
            <span class="mt-2 text-sm font-light text-gray-500"> Copyright © 2020 - 2021 <a href="/" class="mx-2 text-wickedblue hover:text-gray-500" rel="noopener noreferrer">@AlignIt</a>. Since 2021 </span>
          </div>
        </div>
      </footer>

      <!-- Footer Ends -->

      <script src="https://unpkg.com/aos@next/dist/aos.js"></script>
  <script>
    AOS.init();
  </script>
  </body>
</html>
