<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Documentation</title>
    <link href="https://unpkg.com/tailwindcss@^1.0/dist/tailwind.min.css" rel="stylesheet" />
    <link rel="stylesheet" href="https://unpkg.com/aos@next/dist/aos.css" />
    <style>
        html {
            scroll-behavior: smooth;
        }
    </style>
</head>

<body>
    <!-- Navigation begins  -->

    <header class="text-gray-600 body-font">
        <div class="container mx-auto flex flex-wrap p-5 flex-col md:flex-row items-center">
            <a class="flex title-font font-medium items-center text-gray-900 mb-4 md:mb-0" href="https://kdsinghcoder.github.io/Align-it">
                <svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="140" height="90">
                  <path fill="#0c7ecf" d="M21.786,80.958l17.784-48.048l7.8-0.39l17.316,48.438l6.474,1.482l0,4.68l-22.62,0l0-4.68l7.254-1.482l-2.964-8.658l-20.28,0l-3.042,8.658l7.332,1.482l0,4.68l-21.45,0l0-4.68z M50.724,66.06l-7.878-23.478l-8.19,23.478l16.068,0z M87.0408,87.12l0-4.68l7.8-1.56l0-42.12l-10.14,0l-1.95,7.8l-5.07,0l0-14.04l42.51,0l0,14.04l-5.07,0l-1.95-7.8l-10.14,0l0,42.12l7.8,1.56l0,4.68l-23.79,0z" viewBox="0 0 24 24"/>
                  <path fill="#" d=""/></svg>
    
                <span class="ml-3 text-xl">- > Align-it</span>
              </a>
            <nav class="md:ml-auto md:mr-auto flex flex-wrap items-center text-base justify-center">
                <a class="mr-5 hover:text-gray-900"></a>
                <a class="mr-5 hover:text-gray-900"></a>
                <a class="mr-5 hover:text-gray-900"></a>
                <a class="mr-5 hover:text-gray-900"></a>
            </nav>
            <a href="../index.html">
                <button
                    class="inline-flex items-center bg-gray-100 border-0 py-1 px-3 focus:outline-none hover:bg-gray-200 rounded text-base mt-4 md:mt-0">Home
                    Page
                    <svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round"
                        stroke-width="2" class="w-4 h-4 ml-1" viewBox="0 0 24 24">
                        <path d="M5 12h14M12 5l7 7-7 7"></path>
                    </svg>
                </button>
            </a>
        </div>
    </header>
    <!-- Navigation ends -->


    <!-- Content begins -->

    
    <div class="container w-full px-5 py-24 mx-auto lg:px-32">
        <div class="flex flex-col w-full mx-auto mb-2 prose text-left prose-md">
          <div class="mb-5 border-b border-gray-200">
            <div class="flex flex-wrap items-baseline -mt-2">
              <h5>30-06-2021</h5>
              <p class="mt-1 ml-2">The Align Library.</p>
            </div>
          </div>
          <div class="aspect-w-16 aspect-h-9 mx-auto my-5">
            <iframe width="640" height="360" src="https://www.youtube.com/embed/dQXLyeVQrMo" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
          </div>
                <h1 class="mb-2 text-4xl font-extrabold leading-snug tracking-tight text-gray-800 md:text-5xl text-gray-900 hover:text-purple-700"> Sequence Alignment and Smith-Waterman's Contribution. </h1>
           <p  class="mb-4 text-base font-normal text-gray-600"> In 1970, Saul B. Needleman and Christian D. Wunsch proposed a heuristic homology algorithm for sequence alignment, also referred to as the Needleman–Wunsch algorithm.It is a global alignment algorithm that requires {\displaystyle O(mn)}O(mn) calculation steps ({\displaystyle m}m and {\displaystyle n}n are the lengths of the two sequences being aligned). It uses the iterative calculation of a matrix for the purpose of showing global alignment. </p>
          <p  class="mb-4 text-base font-normal text-gray-600"> In recent years, genome projects conducted on a variety of organisms generated massive amounts of sequence data for genes and proteins, which requires computational analysis. Sequence alignment shows the relations between genes or between proteins, leading to a better understanding of their homology and functionality.</p>
          <p  class="mb-4 text-base font-normal text-gray-600"> Smith–Waterman algorithm aligns two sequences by matches/mismatches (also known as substitutions), insertions, and deletions. Both insertions and deletions are the operations that introduce gaps, which are represented by dashes. </p>
          <h1 class="mb-2 text-lg font-extrabold leading-snug tracking-tight text-gray-800 md:text-lg text-gray-900 hover:text-purple-700">Steps - </h1>
          <p  class="mb-4 text-base font-normal text-gray-600">Determine the substitution matrix and the gap penalty scheme. A substitution matrix assigns each pair of bases or amino acids a score for match or mismatch. Usually matches get positive scores, whereas mismatches get relatively lower scores. A gap penalty function determines the score cost for opening or extending gaps. It is suggested that users choose the appropriate scoring system based on the goals. In addition, it is also a good practice to try different combinations of substitution matrices and gap penalties.
Initialize the scoring matrix. The dimensions of the scoring matrix are 1 + length of each sequence respectively. All the elements of the first row and the first column are set to 0. The extra first row and first column make it possible to align one sequence to another at any position, and setting them to 0 makes the terminal gap free from penalty.
Scoring. </p>
          <h3 class="mb-2 text-md font-extrabold leading-snug tracking-tight text-gray-800 md:text-md text-gray-900 hover:text-purple-700"></h3>
          <p  class="mb-4 text-base font-normal text-gray-600">Score each element from left to right, top to bottom in the matrix, considering the outcomes of substitutions (diagonal scores) or adding gaps (horizontal and vertical scores). If none of the scores are positive, this element gets a 0. Otherwise the highest score is used and the source of that score is recorded.
Traceback. </p>
<p  class="mb-4 text-base font-normal text-gray-600">Starting at the element with the highest score, traceback based on the source of each score recursively, until 0 is encountered. The segments that have the highest similarity score based on the given scoring system is generated in this process. To obtain the second best local alignment, apply the traceback process starting at the second highest score outside the trace of the best alignment.</p>
        </div>
      </div>
    

    <!-- Content Ends -->
    
       <!-- Footer starts -->
      
       <footer class="bg-white" aria-labelledby="footer-heading">
        <h2 id="footer-heading" class="sr-only">Footer</h2>
        <div class="px-4 py-12 mx-auto bg-gray-50 max-w-7xl sm:px-6 lg:px-16">
          <div class="flex flex-wrap items-baseline lg:justify-center">
            <span class="mt-2 text-sm font-light text-gray-500"> Copyright © 2020 - 2021 <a href="/" class="mx-2 text-wickedblue hover:text-gray-500" rel="noopener noreferrer">@AlignIt</a>. Since 2021 </span>
          </div>
        </div>
      </footer>

      <!-- Footer Ends -->

      <script src="https://unpkg.com/aos@next/dist/aos.js"></script>
  <script>
    AOS.init();
  </script>
  </body>
</html>
