<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Documentation</title>
    <link href="https://unpkg.com/tailwindcss@^1.0/dist/tailwind.min.css" rel="stylesheet" />
    <link rel="stylesheet" href="https://unpkg.com/aos@next/dist/aos.css" />
    <style>
        html {
            scroll-behavior: smooth;
        }
    </style>
</head>

<body>
    <!-- Navigation begins  -->

    <header class="text-gray-600 body-font">
        <div class="container mx-auto flex flex-wrap p-5 flex-col md:flex-row items-center">
            <a class="flex title-font font-medium items-center text-gray-900 mb-4 md:mb-0" href="https://kdsinghcoder.github.io/Align-it">
                <svg version="1.1" xmlns="http://www.w3.org/2000/svg" width="140" height="90">
                  <path fill="#0c7ecf" d="M21.786,80.958l17.784-48.048l7.8-0.39l17.316,48.438l6.474,1.482l0,4.68l-22.62,0l0-4.68l7.254-1.482l-2.964-8.658l-20.28,0l-3.042,8.658l7.332,1.482l0,4.68l-21.45,0l0-4.68z M50.724,66.06l-7.878-23.478l-8.19,23.478l16.068,0z M87.0408,87.12l0-4.68l7.8-1.56l0-42.12l-10.14,0l-1.95,7.8l-5.07,0l0-14.04l42.51,0l0,14.04l-5.07,0l-1.95-7.8l-10.14,0l0,42.12l7.8,1.56l0,4.68l-23.79,0z" viewBox="0 0 24 24"/>
                  <path fill="#" d=""/></svg>
    
                <span class="ml-3 text-xl">- > Align-it</span>
              </a>
            <nav class="md:ml-auto md:mr-auto flex flex-wrap items-center text-base justify-center">
                <a class="mr-5 hover:text-gray-900"></a>
                <a class="mr-5 hover:text-gray-900"></a>
                <a class="mr-5 hover:text-gray-900"></a>
                <a class="mr-5 hover:text-gray-900"></a>
            </nav>
            <a href="../index.html">
                <button
                    class="inline-flex items-center bg-gray-100 border-0 py-1 px-3 focus:outline-none hover:bg-gray-200 rounded text-base mt-4 md:mt-0">Home
                    Page
                    <svg fill="none" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round"
                        stroke-width="2" class="w-4 h-4 ml-1" viewBox="0 0 24 24">
                        <path d="M5 12h14M12 5l7 7-7 7"></path>
                    </svg>
                </button>
            </a>
        </div>
    </header>
    <!-- Navigation ends -->


    <!-- Content begins -->

    
    <div class="container w-full px-5 py-24 mx-auto lg:px-32">
        <div class="flex flex-col w-full mx-auto mb-2 prose text-left prose-md">
          <div class="mb-5 border-b border-gray-200">
            <div class="flex flex-wrap items-baseline -mt-2">
              <h5>22-05-2020</h5>
              <p class="mt-1 ml-2">The Align Library.</p>
            </div>
          </div>
          <div class="aspect-w-16 aspect-h-9 mx-auto my-5">
            <iframe width="640" height="360" src="https://www.youtube.com/embed/LhpGz5--isw" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
          </div>
                <h1 class="mb-2 text-4xl font-extrabold leading-snug tracking-tight text-gray-800 md:text-5xl text-gray-900 hover:text-purple-700"> Sequence Alignment and Needleman-Wunsch's Contribution. </h1>
          <p  class="mb-4 text-base font-normal text-gray-600">The Needleman–Wunsch algorithm is an algorithm used in bioinformatics to align protein or nucleotide sequences. It was one of the first applications of dynamic programming to compare biological sequences. The algorithm was developed by Saul B. Needleman and Christian D. Wunsch and published in 1970.  </p>
          <p  class="mb-4 text-base font-normal text-gray-600">The algorithm essentially divides a large problem (e.g. the full sequence) into a series of smaller problems, and it uses the solutions to the smaller problems to find an optimal solution to the larger problem. </p>
          <p  class="mb-4 text-base font-normal text-gray-600"> The original purpose of the algorithm described by Needleman and Wunsch was to find similarities in the amino acid sequences of two proteins. Needleman and Wunsch describe their algorithm explicitly for the case when the alignment is penalized solely by the matches and mismatches, and gaps have no penalty (d=0).  </p>
          <h1 class="mb-2 text-lg font-extrabold leading-snug tracking-tight text-gray-800 md:text-lg text-gray-900 hover:text-purple-700">Filling in the table -</h1>
          <p  class="mb-4 text-base font-normal text-gray-600">Start with a zero in the second row, second column. Move through the cells row by row, calculating the score for each cell. The score is calculated by comparing the scores of the cells neighboring to the left, top or top-left (diagonal) of the cell and adding the appropriate score for match, mismatch or indel. Calculate the candidate scores for each of the three possibilities:

The path from the top or left cell represents an indel pairing, so take the scores of the left and the top cell, and add the score for indel to each of them.
The diagonal path represents a match/mismatch, so take the score of the top-left diagonal cell and add the score for match if the corresponding bases (letters) in the row and column are matching or the score for mismatch if they do not.
The resulting score for the cell is the highest of the three candidate scores.

Given there is no 'top' or 'top-left' cells for the second row only the existing cell to the left can be used to calculate the score of each cell. Hence −1 is added for each shift to the right as this represents an indel from the previous score. This results in the first row being 0, −1, −2, −3, −4, −5, −6, −7. The same applies to the first column as only the existing score above each cell can be used.</p>
        <h1 class="mb-2 text-lg font-extrabold leading-snug tracking-tight text-gray-800 md:text-lg text-gray-900 hover:text-purple-700">Backtracking - </h1>
        <p  class="mb-4 text-base font-normal text-gray-600"> Mark a path from the cell on the bottom right back to the cell on the top left by following the direction of the arrows. From this path, the sequence is constructed by these rules:

A diagonal arrow represents a match or mismatch, so the letter of the column and the letter of the row of the origin cell will align.
A horizontal or vertical arrow represents an indel. Vertical arrows will align a gap ("-") to the letter of the row (the "side" sequence), horizontal arrows will align a gap to the letter of the column (the "top" sequence).
If there are multiple arrows to choose from, they represent a branching of the alignments. If two or more branches all belong to paths from the bottom right to the top left cell, they are equally viable alignments. In this case, note the paths as separate alignment candidates. </p>
        </div>
      </div>
    

    <!-- Content Ends -->
    
       <!-- Footer starts -->
      
       <footer class="bg-white" aria-labelledby="footer-heading">
        <h2 id="footer-heading" class="sr-only">Footer</h2>
        <div class="px-4 py-12 mx-auto bg-gray-50 max-w-7xl sm:px-6 lg:px-16">
          <div class="flex flex-wrap items-baseline lg:justify-center">
            <span class="mt-2 text-sm font-light text-gray-500"> Copyright © 2020 - 2021 <a href="/" class="mx-2 text-wickedblue hover:text-gray-500" rel="noopener noreferrer">@AlignIt</a>. Since 2021 </span>
          </div>
        </div>
      </footer>

      <!-- Footer Ends -->

      <script src="https://unpkg.com/aos@next/dist/aos.js"></script>
  <script>
    AOS.init();
  </script>
  </body>
</html>
